src/solvers/smt2_incremental/encoding/struct_encoding.cpp

```diff
@@ -277,3 +277,26 @@ exprt struct_encodingt::decode(
   }
   return simplify_expr(struct_exprt{encoded_fields, original_type}, ns);
 }
+
+exprt struct_encodingt::decode(
+  const exprt &encoded,
+  const union_tag_typet &original_type) const
+{
+  INVARIANT(
+    can_cast_type<bv_typet>(encoded.type()),
+    "Unions are expected to be encoded into bit vectors.");
+  const union_typet definition = ns.get().follow_tag(original_type);
+  const auto &components = definition.components();
+  if(components.empty())
+    return empty_union_exprt{original_type};
+  const auto &component_for_definition = components[0];
```

Comment:
❓ Why only the first component?
What about the others?

Reply:
This matches the SAT backend. I should add a comment to that effect. The return expression is a `union_exprt` which only supports a single member, which makes sense as they could otherwise be in conflict. There are other things we could potentially do or return, but I wanted to aim for consistency; at least initially.